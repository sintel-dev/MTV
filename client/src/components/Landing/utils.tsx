export const colorSchemes = {
  tag: [
    '#FFCD00', // investigate
    '#7CA8FF', // do not investigate
    '#A042FF', // postpone
    '#F64242', // problem
    '#F5FF00', // previously seen
    '#45F642', // normal
    '#C7C7C7', // untagged
  ],
};

// @TODO - later investigation for refactoring (fromIDToTag/fromTagToId)
export const tagSeq = [
  'Investigate',
  'Do not Investigate',
  'Postpone',
  'Problem',
  'Previously seen',
  'Normal',
  'Untagged',
];
const tagClassNames = ['investigate', 'not_investigate', 'postpone', 'problem', 'seen', 'normal', 'untagged'];

export const fromTagToClassName = (tag: string): string => {
  switch (tag) {
    case tagSeq[0]:
      return tagClassNames[0];
    case tagSeq[1]:
      return tagClassNames[1];
    case tagSeq[2]:
      return tagClassNames[2];
    case tagSeq[3]:
      return tagClassNames[3];
    case tagSeq[4]:
      return tagClassNames[4];
    case tagSeq[5]:
      return tagClassNames[5];
    default:
      return 'untagged';
  }
};

export const fromIDtoTag = (id: string): string => {
  switch (id) {
    case '1':
      return tagSeq[0];
    case '2':
      return tagSeq[1];
    case '3':
      return tagSeq[2];
    case '4':
      return tagSeq[3];
    case '5':
      return tagSeq[4];
    case '6':
      return tagSeq[5];
    default:
      return 'untagged';
  }
};

export const fromTagToID = (tag: string): string => {
  switch (tag) {
    case tagSeq[0]:
      return '1';
    case tagSeq[1]:
      return '2';
    case tagSeq[2]:
      return '3';
    case tagSeq[3]:
      return '4';
    case tagSeq[4]:
      return '5';
    case tagSeq[5]:
      return '6';
    default:
      return 'untagged';
  }
};

export const getTagColor = (tag: string): string => {
  let colorIdx: number | undefined;
  for (let i = 0; i < tagSeq.length; i += 1) {
    if (tagSeq[i] === tag) {
      colorIdx = i;
    }
  }
  if (colorIdx === undefined) {
    colorIdx = 6;
  }
  return colorSchemes.tag[colorIdx];
};
