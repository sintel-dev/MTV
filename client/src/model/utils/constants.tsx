export const API_URL = 'http://127.0.0.1:3000/api/v1/';
export const GOOGLE_CLIEN_ID = '796048243680-341g672umjji4m4okfgppm2ogfkrh8mj.apps.googleusercontent.com';
export const SESSION_TOKEN = 'AUTHORIZATION';
