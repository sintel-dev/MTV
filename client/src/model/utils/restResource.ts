import { AxiosRequestConfig, AxiosInstance } from 'axios';

export default class Resource<T, R> {
  /**
   *
   * @param server AxiosInstance
   * @param url Recourse end point
   */
  constructor(private server: AxiosInstance, private url: string) {}

  /**
   * Get all items
   * @param data body data - in most cases, this should be {}
   * @param params URL parameters
   */
  public all(data = {}, params = {}): Promise<R> {
    const promise = this.server.get<R>(this.url, { data, params });
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Get one item
   * @param id the item ID (name) to be retrieved
   * @param data body data â€“ in most cases, this should be {}
   * @param params URL parameters
   */
  public find(id, data = {}, params = {}): Promise<T> {
    const newUrl = `${this.url}${id}`;
    const promise = this.server.get<T>(newUrl, { data, params });
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Create an item.
   * @param data the item data to create
   * @param params URL parameters
   */
  public create(data = {}, params = {}): Promise<T> {
    const promise = this.server.post<T>(this.url, data, { params });
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Delete an item.
   * @param id the item ID (name) to delete
   * @returns Promise of the response's status
   */
  public delete(id): Promise<number> {
    const newUrl = `${this.url}${id}/`;
    const promise = this.server.delete<number>(newUrl);
    return promise
      .then((res) => res.status)
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Update an item.
   * @param id the item ID (name) to update
   * @param data the updated data of this item
   * @param params URL parameters
   */
  public update(id, data = {}, params = {}): Promise<T> {
    const newUrl = `${this.url}${id}/`;
    const promise = this.server.put<T>(newUrl, data, { params });
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }

  /**
   * Allow customize your request.
   * @param config request config
   * @param id item id
   */
  public request<Z>(config: AxiosRequestConfig = {}, id?: string): Promise<Z> {
    let newUrl = this.url;
    if (id !== undefined) {
      newUrl += `${id}/`;
    }
    const promise = this.server.get<Z>(newUrl, config);
    return promise
      .then((res) => {
        if (res.status !== 204) {
          return res.data;
        }
        return null;
      })
      .catch((err) => {
        console.error(err);
        throw err;
      });
  }
}
