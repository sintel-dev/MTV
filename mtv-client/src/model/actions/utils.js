const baseUrl = '/api/v1/';

export const api = {
    get(url, dataType = 'json') {
        const apiUrl = `${baseUrl}${url}`;
        const promise = fetch(apiUrl, {
            mode: 'cors',
            headers: {
                'Access-Control-Allow-Origin': '*',
            },
        });
        return promise.then(response => {
            if (response.status !== 204) {
                return dataType !== 'json' ? response.text : response.json();
            }
            return null;
        })
        .catch(err => console.error(err));
    },

    // post(url, data) {
    //     const apiUrl = `${baseUrl}${url}`;
    //     return fetch(apiUrl, {
    //         mode: 'cors',
    //         method: 'POST',
    //         headers: {
    //             'Access-Control-Allow-Origin': '*',
    //             'Content-Type': 'application/json',
    //             // 'Authorization': lscache.get('authorization')
    //         },
    //         body: JSON.stringify(data)
    //     });
    // }
};
