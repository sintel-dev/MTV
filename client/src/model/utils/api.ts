import axios, { AxiosRequestConfig, AxiosInstance } from 'axios';
import Cookies from 'js-cookie';
import { API_URL, SESSION_TOKEN } from './constants';
// import { SESSION_TOKEN } from '@nx-react/session';

import Resource from './restResource';
import {
  DatasetDataType,
  DatasetsResponse,
  DatarunDataType,
  DatarunsResponse,
  ExperimentDataType,
  ExperimentsResponse,
  PipelinesResponse,
  EventsResponse,
  CommentsResponse,
  DataResponse,
} from '../types/index';
import { PipelineDataType } from '../types/pipeline';
import { EventDataType } from '../types/event';
import { CommentDataType } from '../types/comment';

export class RestClient {
  private server: AxiosInstance;

  // default config for http request
  private requestConfig: AxiosRequestConfig = {
    baseURL: API_URL,
    // timeout: 2000,
  };

  public test: Resource<any, any>;

  public datasets: Resource<DatasetDataType, DatasetsResponse>;

  public dataruns: Resource<DatarunDataType, DatarunsResponse>;

  public experiments: Resource<ExperimentDataType, ExperimentsResponse>;

  public pipelines: Resource<PipelineDataType, PipelinesResponse>;

  public events: Resource<EventDataType, EventsResponse>;

  public comments: Resource<CommentDataType, CommentsResponse>;

  public data: Resource<any, DataResponse>;

  /**
   *
   * @param config AxiosRequestConfig
   */
  constructor(config: AxiosRequestConfig) {
    this.requestConfig = { ...this.requestConfig, ...config };
    this.server = axios.create(this.requestConfig);

    // add resources
    this.test = new Resource(this.server, 'test/');
    this.datasets = new Resource(this.server, 'datasets/');
    this.dataruns = new Resource(this.server, 'dataruns/');
    this.experiments = new Resource(this.server, 'experiments/');
    this.pipelines = new Resource(this.server, 'pipelines/');
    this.events = new Resource(this.server, 'events/');
    this.comments = new Resource(this.server, 'comments/');
    this.data = new Resource(this.server, 'data/');
    this.server.interceptors.request.use(this.requestInterceptor);
    this.server.interceptors.response.use(this.responseSuccessInterceptor, this.responseFailInterceptor);
  }

  public requestInterceptor(config) {
    const authHeader = Cookies.get(SESSION_TOKEN);
    if (!authHeader) {
      return config;
    }

    return {
      ...config,
      headers: {
        ...config.headers,
        Authorization: authHeader,
      },
    };
  }

  public responseSuccessInterceptor(response) {
    return response;
  }

  public responseFailInterceptor(error) {
    if (error.response.status === 401) {
      window.location.href = '/logout';
    }

    return Promise.reject(error);
  }
}

export default new RestClient({
  baseURL: 'http://127.0.0.1:3000/api/v1/',
});
