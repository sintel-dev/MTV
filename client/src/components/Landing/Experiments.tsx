import React from 'react';
import { connect } from 'react-redux';
import * as _ from 'lodash';
import { withRouter, useHistory } from 'react-router-dom';
import Loader from '../Common/Loader';
import {
  getFilteredExperiments,
  getIsExperimentsLoading,
  getSelectedPipeline,
  getSelectedExperiment,
} from '../../model/selectors/projects';
import { selectExperiment } from '../../model/actions/landing';
import { RootState, ExperimentDataType } from '../../model/types';
import { TagStats, Scale as MatrixScale } from './Matrix/types';
import { fromTagToID } from './utils';
import Matrix from './Matrix/Matrix';

let props: Props;
type StateProps = ReturnType<typeof mapState>;
type DispatchProps = ReturnType<typeof mapDispatch>;
type Props = StateProps & DispatchProps;
type renderExperimentProps = {
  experiment: ExperimentDataType;
  tagStats: TagStats;
  matrixScale: MatrixScale;
  index: number;
  selectedPipeline: typeof props.selectedExperiment;
  selectedExperiment: typeof props.selectedExperiment;
  onSelectExperiment: typeof props.onSelectExperiment;
};

const Experiments: React.FC<Props> = ({
  isExperimentsLoading,
  filteredExperiments,
  onSelectExperiment,
  selectedPipeline,
  selectedExperiment,
}) => {
  // Compute maxTagNum, maxEventNum, and maxScore
  // which would be used for plotting Matrix

  let maxTagNum = Number.MIN_SAFE_INTEGER;
  let maxEventNum = Number.MIN_SAFE_INTEGER;
  let maxScore = Number.MIN_SAFE_INTEGER;
  let tagStatsList: TagStats[] = [];

  // @TODO - move this logic to selectors/experiment
  _.each(filteredExperiments, (experiment) => {
    let tagStats: { [index: string]: number } = {};
    for (let i = 0; i < 7; i += 1) {
      tagStats[String(i)] = 0;
    }
    _.each(experiment.dataruns, (datarun) => {
      for (let i = 0; i < datarun.events.length; i += 1) {
        let tid = fromTagToID(datarun.events[i].tag);
        tid = tid === 'untagged' ? '0' : tid;
        if (!_.has(tagStats, tid)) {
          tagStats[tid] = 0;
        }
        tagStats[tid] += 1;
        maxTagNum = maxTagNum < tagStats[tid] ? tagStats[tid] : maxTagNum;
        maxScore = maxScore > datarun.events[i].score ? maxScore : datarun.events[i].score;
        maxEventNum = maxEventNum < datarun.events.length ? datarun.events.length : maxEventNum;
      }
    });
    tagStatsList.push(tagStats);
  });

  const matrixScale: MatrixScale = {
    maxTagNum,
    maxEventNum,
    maxScore,
  };

  return (
    <div className="item-row scroll-style" id="experiments" data-name="experiments">
      <h2>Experiments</h2>
      <div className="item-wrapper">
        <Loader isLoading={isExperimentsLoading}>
          {filteredExperiments.length ? (
            filteredExperiments.map((experiment, index) => {
              const experimentProps = {
                experiment,
                tagStats: tagStatsList[index],
                matrixScale,
                index,
                onSelectExperiment,
                selectedPipeline,
                selectedExperiment,
              };
              return <Experiment key={experiment.id} {...experimentProps} />;
            })
          ) : (
            <h2>No experiments found</h2>
          )}
        </Loader>
      </div>
    </div>
  );
};

const countDatarunEvents = (experiment) => {
  const { dataruns } = experiment;
  return dataruns.map((datarun) => datarun.events.length).reduce((item, accumulator) => item + accumulator, 0);
};

const Experiment: React.FC<renderExperimentProps> = ({
  experiment,
  tagStats,
  matrixScale,
  index,
  selectedPipeline,
  selectedExperiment,
}) => {
  let history = useHistory();
  const activeClass = selectedPipeline || selectedExperiment === experiment.id ? 'active' : '';
  const eventCounts = countDatarunEvents(experiment);
  return (
    <div className={`cell ${activeClass}`} key={index} onClick={() => history.push(`/experiment/${experiment.id}`)}>
      <h3>
        #{index + 1} {experiment.dataset}_{experiment.pipeline}
      </h3>
      <div className="item-data">
        <ul>
          <li>Signals: {experiment.dataruns.length}</li>
          <li>Events: {eventCounts}</li>
          <li>DC: {experiment.date_creation.substring(0, 10)}</li>
          <li>By: {`${experiment.created_by}`}</li>
        </ul>
        <Matrix experiment={experiment} tagStats={tagStats} scale={matrixScale} />
      </div>
    </div>
  );
};

const mapState = (state: RootState) => ({
  filteredExperiments: getFilteredExperiments(state),
  isExperimentsLoading: getIsExperimentsLoading(state),
  selectedPipeline: getSelectedPipeline(state),
  selectedExperiment: getSelectedExperiment(state),
});

const mapDispatch = (dispatch: Function) => ({
  onSelectExperiment: (history, experiment: string) => dispatch(selectExperiment(history, experiment)),
});

export default withRouter(connect<StateProps, DispatchProps, {}, RootState>(mapState, mapDispatch)(Experiments));
