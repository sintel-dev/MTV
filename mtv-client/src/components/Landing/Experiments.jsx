import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Loader from '../Common/Loader';
import {
  getFilteredExperiments,
  getIsExperimentsLoading,
  getSelectedPipeline,
  getSelectedExperiment,
} from '../../model/selectors/projects';
import { selectExperiment } from '../../model/actions/landing';


const renderExperiment = (experiment, index, onSelectExperiment, selectedPipeline, selectedExperiment) => {
  const activeClass = selectedPipeline || selectedExperiment === experiment.id ? 'active' : '';
  return (
    <div className={`cell ${activeClass}`} key={index} onClick={() => onSelectExperiment(experiment.id)}>
      <h3>#{index + 1} {experiment.dataset}_{experiment.pipeline}</h3>
      <div className="item-data">
        <ul>
          <li>Signals: {experiment.dataruns.length}</li>
          <li>DC: {experiment.date_creation.substring(0, 10)}</li>
        </ul>
      </div>
    </div>);
};

const Experiments = ({
  isExperimentsLoading,
  filteredExperiments,
  onSelectExperiment,
  selectedPipeline,
  selectedExperiment,
}) => (
  <div className="item-row scroll-style" id="experiments">
    <h2>Experiments</h2>
    <div className="item-wrapper">
      <Loader isLoading={isExperimentsLoading}>
        {
          filteredExperiments.length ?
            filteredExperiments.map((experiment, index) =>
            renderExperiment(experiment, index, onSelectExperiment, selectedPipeline, selectedExperiment)) :
            <h2>No experiments found</h2>
          }
      </Loader>
    </div>
  </div>);

Experiments.propTypes = {
  filteredExperiments: PropTypes.array,
  isExperimentsLoading: PropTypes.bool,
  onSelectExperiment: PropTypes.func,
  selectedPipeline: PropTypes.string,
  selectedExperiment: PropTypes.string,
};

export default connect(state => ({
  filteredExperiments: getFilteredExperiments(state),
  isExperimentsLoading: getIsExperimentsLoading(state),
  selectedPipeline: getSelectedPipeline(state),
  selectedExperiment: getSelectedExperiment(state),
}), dispatch => ({
  onSelectExperiment: (experiment) => dispatch(selectExperiment(experiment)),
}))(Experiments);
