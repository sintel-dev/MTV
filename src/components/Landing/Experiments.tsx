import React from 'react';
import { connect } from 'react-redux';
import * as _ from 'lodash';
import { withRouter, useHistory } from 'react-router-dom';
import { RootState, ExperimentDataType } from 'src/model/types';
import { selectExperiment } from 'src/model/actions/landing';
import {
  getFilteredExperiments,
  getIsExperimentsLoading,
  getSelectedPipeline,
  getSelectedExperiment,
} from 'src/model/selectors/projects';
import Loader from '../Common/Loader';
import { TagStats, MatrixScale } from './Matrix/types';
import { fromTagToID } from './utils';
import Matrix from './Matrix/Matrix';

type StateProps = ReturnType<typeof mapState>;
type DispatchProps = ReturnType<typeof mapDispatch>;
type Props = StateProps & DispatchProps;

let props: Props;
type ExperimentProps = {
  experiment: ExperimentDataType;
  tagStats: TagStats;
  matrixScale: MatrixScale;
  index: number;
  selectedPipeline: typeof props.selectedPipeline;
  selectedExperiment: typeof props.selectedExperiment;
};

const countDatarunEvents = (experiment) => {
  const { dataruns } = experiment;
  return dataruns.map((datarun) => datarun.events.length).reduce((item, accumulator) => item + accumulator, 0);
};

export const Experiment: React.FC<ExperimentProps> = ({
  experiment,
  tagStats,
  matrixScale,
  index,
  selectedPipeline,
  selectedExperiment,
}: ExperimentProps) => {
  let history = useHistory();
  const activeClass = selectedPipeline || selectedExperiment === experiment.id ? 'active' : '';
  const eventCounts = countDatarunEvents(experiment);
  const { id, dataruns, name, date_creation, created_by } = experiment;

  return (
    <div className={`cell ${activeClass}`} key={index} onClick={() => history.push(`/experiment/${id}`)}>
      <h3>
        #{index + 1} {name}
      </h3>
      <div className="item-data">
        <ul>
          <li>Signals: {dataruns.length}</li>
          <li>Events: {eventCounts}</li>
          <li>DC: {date_creation.substring(0, 10)}</li>
          <li>By: {`${created_by}`}</li>
        </ul>
        <Matrix experiment={experiment} tagStats={tagStats} scale={matrixScale} />
      </div>
    </div>
  );
};

export const Experiments: React.FC<Props> = ({
  isExperimentsLoading,
  filteredExperiments,
  onSelectExperiment,
  selectedPipeline,
  selectedExperiment,
}: Props) => {
  // Compute maxTagNum, maxEventNum, and maxScore
  // which would be used for plotting Matrix
  let maxTagNum: number = Number.MIN_SAFE_INTEGER;
  let maxEventNum: number = Number.MIN_SAFE_INTEGER;
  let maxScore: number = Number.MIN_SAFE_INTEGER;
  let tagStatsList: TagStats[] = [];

  filteredExperiments.forEach((currentExperiment) => {
    const { dataruns } = currentExperiment;
    let tagStats: { [index: string]: number } = {};

    for (let i = 0; i < 7; i += 1) {
      tagStats[String(i)] = 0;
    }

    dataruns.forEach((currentDatarun) => {
      const { events } = currentDatarun;
      for (let iterator = 0; iterator < events.length; iterator += 1) {
        let tagID: string = fromTagToID(events[iterator].tag);

        tagID = tagID === 'untagged' ? '0' : tagID;
        if (!_.has(tagStats, tagID)) {
          tagStats[tagID] = 0;
        }

        tagStats[tagID] += 1;
        maxTagNum = maxTagNum < tagStats[tagID] ? tagStats[tagID] : maxTagNum;
        maxScore = maxScore > events[iterator].score ? maxScore : events[iterator].score;
        maxEventNum = maxEventNum < events.length ? events.length : maxEventNum;
      }
    });
    tagStatsList.push(tagStats);
  });

  const matrixScale: MatrixScale = {
    maxTagNum,
    maxEventNum,
    maxScore,
  };

  return (
    <div className="item-row scroll-style" id="experiments" data-name="experiments">
      <h2>Experiments</h2>
      <div className="item-wrapper">
        <Loader isLoading={isExperimentsLoading}>
          {filteredExperiments.length ? (
            filteredExperiments.map((experiment, index) => {
              const experimentProps = {
                experiment,
                tagStats: tagStatsList[index],
                matrixScale,
                index,
                onSelectExperiment,
                selectedPipeline,
                selectedExperiment,
              };
              return <Experiment key={experiment.id} {...experimentProps} />;
            })
          ) : (
            <h2>No experiments found</h2>
          )}
        </Loader>
      </div>
    </div>
  );
};

const mapState = (state: RootState) => ({
  filteredExperiments: getFilteredExperiments(state),
  isExperimentsLoading: getIsExperimentsLoading(state),
  selectedPipeline: getSelectedPipeline(state),
  selectedExperiment: getSelectedExperiment(state),
});

const mapDispatch = (dispatch: Function) => ({
  onSelectExperiment: (history, experiment: string) => dispatch(selectExperiment(history, experiment)),
});

export default withRouter(connect<StateProps, DispatchProps, {}, RootState>(mapState, mapDispatch)(Experiments));
