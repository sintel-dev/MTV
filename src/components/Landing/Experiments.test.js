import React from 'react';
import { Experiment } from './Experiments';
import { renderWithStore, TestWrapper } from '../../tests/utils';

jest.mock('./Matrix/Matrix', () => () => 'Matrix component');
describe('Testing experiments component ->', () => {
  const experimentsProps = {
    experiment: {
      id: '5da7cc6376e3e19307d0db64',
      project: 'SMAP',
      dataset: 'SMAP_set1',
      date_creation: '2019-10-17T02:05:23.503000',
      created_by: null,
      pipeline: 'lstm',
      dataruns: [
        {
          id: '5da7cc6576e3e19307d0db65',
          signal: 'T-1',
          events: [{ start_time: 1228219200, stop_time: 1229299200, score: 0.0336717714726245, tag: null }],
        },
      ],
    },
    tagStats: { 0: 30, 1: 0, 2: 0, 3: 0, 4: 0, 5: 0, 6: 0 },
    matrixScale: { maxTagNum: 64, maxEventNum: 10, maxScore: 1.5598205662590257 },
    index: 0,
    selectedPipeline: 'cyclegan',
    selectedExperiment: '5da7d1a076e3e19307d0dc19',
  };

  const mountedExperiment = renderWithStore(
    {},
    <TestWrapper>
      <Experiment {...experimentsProps} />,
    </TestWrapper>,
  );

  it('Should render experiments without crashing', () => {
    expect(mountedExperiment).toMatchSnapshot();
  });

  // @TODO - find a way to test history push
  // it('Should open experiment page when selecting experiment', () => {
  //   const mockHistory = { push: jest.fn() };
  //   const mountedExpComponent = mountWithStore(
  //     {},
  //     <TestWrapper>
  //       <Experiment {...experimentsProps} history={mockHistory} />,
  //     </TestWrapper>,
  //   );
  //   const expItem = mountedExpComponent.find('.cell.active');
  //   const spy = jest.spyOn(mockHistory, 'push');

  //   expItem.simulate('click');
  //   expect(spy).toHaveBeenCalled();
  // });
});
