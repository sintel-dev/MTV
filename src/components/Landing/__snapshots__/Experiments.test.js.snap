// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing experiments component -> Should render experiments without crashing 1`] = `
Array [
  <div
    class="cell active"
  >
    <h3>
      #1 
    </h3>
    <div
      class="item-data"
    >
      <ul>
        <li>
          Signals: 1
        </li>
        <li>
          Events: 1
        </li>
        <li>
          DC: 2019-10-17
        </li>
        <li>
          By: null
        </li>
      </ul>
      Matrix component
    </div>
  </div>,
  ",",
]
`;
